proof/refine/ARM/TcbAcc_R.thy

```diff
@@ -4637,27 +4649,24 @@ lemma sbn_st_tcb'[wp]:
 
 crunches rescheduleRequired, tcbSchedDequeue, setThreadState, setBoundNotification, scheduleTCB
   for typ_at'[wp]: "\<lambda>s. P (typ_at' T p s)"
+  and sc_at'_n[wp]: "\<lambda>s. P (sc_at'_n n p s)"
+  and ctes_of[wp]: "\<lambda>s. P (ctes_of s)"
   (wp: crunch_wps)
 
-lemmas rescheduleRequired_typ_ats[wp] = typ_at_lifts[OF rescheduleRequired_typ_at']
-lemmas tcbSchedDequeue_typ_ats[wp] = typ_at_lifts[OF tcbSchedDequeue_typ_at']
-
-lemma setThreadState_ctes_of[wp]:
-  "setThreadState st t \<lbrace>\<lambda>s. P (ctes_of s)\<rbrace>"
-  apply (clarsimp simp: setThreadState_def)
-  by (wpsimp wp: threadSet_ctes_of)
-
-end
-
-lemmas threadSet_typ_at_lifts[wp] = typ_at_lifts [OF threadSet_typ_at']
+global_interpretation rescheduleRequired: typ_at_all_props' "rescheduleRequired"
+  by typ_at_props'
 
-interpretation setThreadState: typ_at_ctes_of_props' "setThreadState st p"
-  by (unfold_locales) (wpsimp wp: setThreadState_typ_at')+
+global_interpretation tcbSchedDequeue: typ_at_all_props' "tcbSchedDequeue thread"
+  by typ_at_props'
 
-interpretation setBoundNotification: typ_at_props' "setBoundNotification v p"
-  by (unfold_locales) (rule setBoundNotification_typ_at')
+global_interpretation threadSet: typ_at_all_props' "threadSet f p"
+  by typ_at_props'
 
+global_interpretation setThreadState: typ_at_all_props' "setThreadState st p"
+  by typ_at_props'
 
+global_interpretation setBoundNotification: typ_at_all_props' "setBoundNotification v p"
```

Comment:
After seeing so many of these, I'm starting to think it'd be nice to have a short-hand command doing many interpretations at the same time (a bit like `crunches`). Problem for some other time, but it might actually be fairly easy to write.

Reply:
Yeah, I thought the same and something like that would definitely have been nice while I was making these changes. This was already enough of a side project though, so I resisted the temptation to implement it.